### PR TITLE
Add device tree support for qemu_xtensa and xt-sim

### PR DESCRIPTION
**Pull request by SavinayDharmappa**

### CODE REVIEW

File: arch/xtensa/soc/sample_controller/linker.ld [hunk 1]
```diff
@@ -48,7 +49,8 @@ MEMORY
   iram0_19_seg :                      	org = 0x40000400, len = 0x1FC00
   srom0_seg :                         	org = 0x50000000, len = 0x300
   srom1_seg :                         	org = 0x50000300, len = 0xFFFD00
-  sram0_seg :                         	org = 0x60000000, len = 0x4000000
+  sram0_seg :                         	org = CONFIG_SRAM_BASE_ADDRESS,
+					len = CONFIG_SRAM_SIZE*1K
```
Comment (andyross) [inline arch/xtensa/soc/sample_controller/linker.ld]: Said this earlier, will repeat: I know this sounds like it's a cool feature to unify memory mapping across all the targets.  It's not.  It's a footgun.  We **will** have bugs due to this decoupling, where systems inexplicably fail due to their memory map being "configured" when it never should have been configurable in the first place.

This is the linker script.  It is the final (and should be the only) source for authority when it comes to device memory layout.  Putting it anywhere else is only going to hurt us.

I'm not going to -1 an in-progress architectural change over this.  But I won't shut up that I think it's a terrible idea.

File: boards/xtensa/qemu_xtensa/qemu_xtensa.dts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/dts-v1/;
+
+#include "sample_controller.dtsi"
+
+/ {
+	model = "qemu_xtensa";
+	compatible = "sample,controller";
```
Comment (andyross) [inline boards/xtensa/qemu_xtensa/qemu_xtensa.dts]: compatible is usually used for driver selection.  Is "sample" or "controller" really unique enough?  I know there's no "driver" for the CPU, but the principle still seems to be violated here.

"sample_controller" is an Xtensa string identifying the CPU that is implemented by qemu-xtensa.  I don't think it's reasonably decomposable into being both "sample" and "controller"

File: boards/xtensa/xt-sim/xt-sim.dts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/dts-v1/;
+
+#include "sample_controller.dtsi"
+
+/ {
+	model = "xt-sim";
+	compatible = "sample,controller";
```
Comment (andyross) [inline boards/xtensa/xt-sim/xt-sim.dts]: Hm... I don't do the cadence tools so much, but I'm pretty sure that "xt-sim" is **not** a "sample_controller".  Those are two different CPU instances (actually xt-sim is a whole family I think).

**Review by andyross — COMMENTED** [review]
Seems harmless enough.  Continued whining about the memory map bug factory.   Not sure I like the choice of "compatible" strings, but AFAICT those are arbitrary and won't break anything.

**Comment by galak** [comment]
@SavinayDharmappa Can you update this once we close on #7581?

**Review by galak — CHANGES_REQUESTED** [review]
Do we need to add xtensa,sim-console?  Since its not a uart or trying to mimic a uart driver?

**Comment by galak** [comment]
> Do we need to add xtensa,sim-console? Since its not a uart or trying to mimic a uart driver?

?

**Comment by SavinayDharmappa** [comment]
> 
> 
> > Do we need to add xtensa,sim-console? Since its not a uart or trying to mimic a uart driver?
> 
> ?

It is not required. I have removed the bindings for and pushed the changes

**Comment by galak** [comment]
> It is not required. I have removed the bindings for and pushed the changes

It still show's up in dts/xtensa/sample_controller.dtsi

**Comment by galak** [comment]
Looks good, can you fix the missing license/copyright in boards/xtensa/qemu_xtensa/qemu_xtensa.dts & boards/xtensa/xt-sim/xt-sim.dts

**Comment by SavinayDharmappa** [comment]
@galak  done.

**Review by galak — APPROVED** [review]

**Pull request merged**